src/coreclr/jit/importercalls.cpp

```diff
@@ -3987,6 +3987,38 @@ GenTree* Compiler::impSRCSUnsafeIntrinsic(NamedIntrinsic        intrinsic,
         {
             assert((sig->sigInst.methInstCount == 1) || (sig->sigInst.methInstCount == 2));
 
+            if (sig->sigInst.methInstCount == 1)
+            {
+                const CORINFO_CLASS_HANDLE inst = sig->sigInst.methInst[0];
+                assert(inst != nullptr);
+
+                if ((info.compCompHnd->getClassAttribs(inst) & CORINFO_FLG_SHAREDINST) == 0)
```

Comment:
This is going to make us skip e.g. `Unsafe.As<List<Canon>>`. We do not want that.

Delete this early out and just depend on `isMoreSpecificType` to see whether the type is more exact.